hecate.py

```diff
@@ -297,56 +407,90 @@ def process(arguments):
                     else:
-                        name = entry + '_encrypted'
-                        result['upload'][name] = runner.cloud_file(name)
-                else:  # skip the file
-                    result['upload'][entry] = 'Skipped due to previous error.'
-            except Exception as e:
-                result['upload'][entry] = 'Error: %s' % str(e)
-                skipFiles.append(entry)
-        if arguments.decrypt:
+                        result['encrypt'][entry] = 'Skipped due to previous error.'
+                except Exception as e:
+                    result['encrypt'][entry] = 'Error: %s' % str(e)
+                    skipFiles.append(entry)
+            if arguments.upload:
+                try:
+                    if entry not in skipFiles:
+                        if arguments.inplace or not result.get('encrypt'):
+                            result['upload'][entry] = runner.cloud_file(entry,
+                                container = arguments.specifyContainer)
+                        # they have not specified inplace but did specify encrypt
+                        else:
+                            name = entry + '_encrypted'
+                            result['upload'][name] = runner.cloud_file(name,
+                                container = arguments.specifyContainer)
+                    else:  # skip the file
+                        result['upload'][entry] = 'Skipped due to previous error.'
+                except Exception as e:
+                    result['upload'][entry] = 'Error: %s' % str(e)
+                    skipFiles.append(entry)
+            if arguments.decrypt:
+                try:
+                    if entry not in skipFiles:
+                        result['decrypt'][entry] = runner.decrypt_file(
+                            entry, arguments.key, arguments.inplace)
+                    else:
+                        result['encrypt'][entry] = 'Skipped due to previous error.'
+                except Exception as e:
+                    result['decrypt'][entry] = 'Error: %s' % str(e)
+                    skipFiles.append(entry)
+    if arguments.remove:
+        for entry in arguments.remove[0]:
             try:
                 if entry not in skipFiles:
-                    result['decrypt'][entry] = runner.decrypt_file(
-                        entry, arguments.key, arguments.inplace)
+                    result['remove'][entry] = runner.cloud_file_remove(
+                        entry, container = arguments.specifyContainer)
                 else:
-                    result['encrypt'][entry] = 'Skipped due to previous error.'
+                    result['remove'][entry] = 'Skipped due to previous error.'
             except Exception as e:
-                result['decrypt'][entry] = 'Error: %s' % str(e)
+                result['remove'][entry] = 'Error: %s' % str(e)
                 skipFiles.append(entry)
+    if arguments.newContainer:
```

Comment:
I suspect this code needs to be run before arguments.upload, otherwise you'll be trying to upload to a container that doesn't exist?